apps/authentication-portal/src/main/webapp/basicauth.jsp

```diff
@@ -257,18 +248,38 @@
 
 <% if (Boolean.parseBoolean(loginFailed) && errorCode.equals(IdentityCoreConstants.USER_ACCOUNT_NOT_CONFIRMED_ERROR_CODE) && request.getParameter("resend_username") == null) { %>
     <div class="ui visible warning message" id="error-msg" data-testid="login-page-error-message">
-        <%= AuthenticationEndpointUtil.i18n(resourceBundle, errorMessage) %>
-
-        <div class="ui divider hidden"></div>
-
-        <%=AuthenticationEndpointUtil.i18n(resourceBundle, "no.confirmation.mail")%>
-
-        <a id="registerLink"
-            href="javascript:showResendReCaptcha();"
-            data-testid="login-page-resend-confirmation-email-link"
-        >
-            <%=StringEscapeUtils.escapeHtml4(AuthenticationEndpointUtil.i18n(resourceBundle, "resend.mail"))%>
-        </a>
+        <form action="login.do?resend_username=<%=Encode.forHtml(URLEncoder.encode(request.getParameter("failedUsername"), UTF_8))%>&<%=AuthenticationEndpointUtil.cleanErrorMessages(Encode.forJava(request.getQueryString()))%>" method="post" id="resendForm">
+            <%= AuthenticationEndpointUtil.i18n(resourceBundle, errorMessage) %>
+
+            <div class="ui divider hidden"></div>
+
+            <%=AuthenticationEndpointUtil.i18n(resourceBundle, "no.confirmation.mail")%>
+
+            <button id="registerLink"
+                style="
+                    padding: 0 !important;
+                    background: none !important;
+                    border: none;
+                    cursor: pointer;
+                    color: #ff5000;
+                "
+                class="g-recaptcha"
+                <%
+                    if (reCaptchaResendEnabled) {
+                %>
+                data-sitekey="<%=Encode.forHtmlContent(reCaptchaKey)%>"
+                <%
+                    }
+                %>
+                data-callback="onSubmitResend"
+                data-action="resendConfirmation"
+                onmouseover='this.style.textDecoration="underline"'
+                onmouseout='this.style.textDecoration="none"'
```

Comment:
I don't think we are underlining any buttons when hovered over. It's better to avoid this here to be consistent.

Reply:
This is a button styled as a <a> tag

Reply:
Recaptcha does not work in a <a> tag, so had to use a button

Reply:
Fixed in https://github.com/wso2/identity-apps/pull/3298